### ISSUE TITLE
Network is not learning. mAP stays the same and Total Loss does not go below 6

### ISSUE THREAD

**Issue by GustavZ** [victordibia/handtracking#5]
Hey Victor, i had the same idea to do a realtime capable hand detector.
First i also did it with the oxford dataset, but like you said it is not really good.
Then i saw your repo and tried the egohands dataset which looks really promising.

But with both datasets my mAp always stays at 0 and the total loss never goes below ~6.
If you look at the weights you see that they stayed around zero, so no learning at all.

Do you have an idea why this could be?

Here is a screenshot of my current training with the exact same config you are using.
(by the way, why do you use a batch size of only 6? Is this any good except of saving memory?)
![screenshot from 2018-01-19 13-21-37](https://user-images.githubusercontent.com/29252883/35158817-96db4858-fd38-11e7-83f6-c393c467fd33.png)
![screenshot from 2018-01-19 13-21-11](https://user-images.githubusercontent.com/29252883/35158829-9d537372-fd38-11e7-8b6a-3c49e2d7a030.png)

I would really appreciate it if you could help me with this.

And yeah, your egohands_dataset_clean.py code is really nice :+1: 

EDIT: Are you sure the csv files that you create contain the correct bounding boxes? Could you share the csv files you used to create the tf record files? Thanks!

**Comment by victordibia** [comment]
Hi @GustavZ 

- Zero mAP
This is quite unusual, and I agree with you that an mAP of 0 suggests there is something wrong with the labelling.
- Batch Size
This was selected only because of memory issues. 

- Original CSV files
I am attaching my initial train/test label csv files, which you can compare to what is generated by the egohands script in the repo. I did some code cleanup after I built the original model so there is a slight possibility something changed slightly.  

[csvs.zip](https://github.com/victordibia/handtracking/files/1647447/csvs.zip)

Something you could do is verifying if training proceeds reasonably using only the tf records before combining it with the oxford dataset.

**Comment by GustavZ** [comment]
Hey Victor,

Thanks for your quick reply!
I will compare the csv files on Monday when I’m back on my machine.

The training I was talking about was done only on the egohands dataset.
So no combining with oxford in this part.

The strange thing I had similar bad results on training solely on the Oxford dataset as this time on the egohands.

But training with all my other datasets runs as expected, that’s why I assumed there is something wrong with the csv files, as my tf 1.4 setup/config as well as my csv to tfrecord script appears to work normal on other sets...

**Comment by GustavZ** [comment]
I compared your csv files with mine and they are totally different.
Yours are perfectly ordered by name and mine are all shuffled.
Furthermore the values are not the same. 
Here is an example for unequal values (see rest attached):
[csvs_gustav.tar.gz](https://github.com/victordibia/handtracking/files/1651264/csvs_gustav.tar.gz)
YOURS:
```
filename,width,height,class,xmin,ymin,xmax,ymax
CARDS_COURTYARD_B_T_frame_0011.jpg,1280,720,hand,647,453,824,551
CARDS_COURTYARD_B_T_frame_0011.jpg,1280,720,hand,515,431,622,543
```

MINE:
```
CARDS_COURTYARD_B_T_frame_0011.jpg,1280,720,hand,803,623,981,718
CARDS_COURTYARD_B_T_frame_0011.jpg,1280,720,hand,679,340,856,464
CARDS_COURTYARD_B_T_frame_0011.jpg,1280,720,hand,511,299,676,428
```


But the thing is, i did not create the csv files by my own, i used your script to create them from the default dataset.

How did you modify your code after the commit zu github? Could you update it?

Would be very thankfull!

**Comment by victordibia** [comment]
Hmmm ... I actually did a similar check on Friday before I sent you my original csv's. I generated train and test csv's using the script and compared them and there were similar values for the same images. Can you retry the process? I'll suggest you pull a **fresh** copy of the repo and start from scratch.
Let me know if you still get different values.
-V.

**Comment by GustavZ** [comment]
I did a complete delete and re-clone of your repo.
But the result is still the same, the csv are still in a shuffled order and tain and test constist of different images with different bboxes than your csv files.

What could be the reason? I am using python 2.7 and openCV 3.3.1 if this could be important.

**Comment by GustavZ** [comment]
Hey Victor,
i still have not found the reason why my machine produces completely different csv files with the same code you are using.
Have you ever experienced something similar with creating csv files?
Could you maybe share your train/test directories including images and the final csv files?

**Comment by GustavZ** [comment]
Are you sure you convert the bounding boxes correctly? 
Where do you set your coordinate center in the image? Top left or Bottom left?

Also i saw that You got some unused variables in your `get_bbox_visualize()` function:
```
index = 0
height = width = 0
```
idk if this was on purpose or forgotten for some value convertation?

**Comment by Yemen-Romanian** [comment]
Hello guys, @victordibia and @gustavz. I have almost the same issue with training my custom hand detector. I use a pre-trained model on EgoHands dataset Victor has provided and I do transfer learning on EgoFinger dataset. This dataset includes hand images taken from egocentric viewpoint. Image size is 640x480, format is png. For testing purposes I train on a little subset. But after every epoch I get zero mAp. 
Also, I tried to learn by taking vanilla SSD and train on my dataset, but result is the same. 
I am pretty sure that I have generated right csv files and tfrecords. On tensorboard "Image" tab I can view visualization of ground truth boxes and they are correct.
Are there any thoughts for what can be done? Thank you in advance!
